pkg/common/ipsec/utils.go

```diff
@@ -18,10 +18,12 @@ const (
 func CountUniqueIPsecKeys(states []netlink.XfrmState) int {
 	keys := make(map[string]bool)
 	for _, s := range states {
-		if s.Aead == nil {
-			continue
+		if s.Aead != nil {
+			keys[string(s.Aead.Key)] = true
+		}
+		if s.Crypt != nil {
+			keys[string(s.Crypt.Key)] = true
 		}
```

Comment:
Shoudn't we be handling the authentication key as well?

Reply:
As I understood in case of CBC we have both:
```
Crypt name: cbc(aes), key: Q���3��90x��, str: {Name: cbc(aes), Key: 0x0151a41da39e3310d4f58b3930788dc4}
Auth name: hmac(sha512), key: �M�������MQ�s�������~M��8��1">�, str: {Name: hmac(sha512), Key: 0x8b4d92bf9396e7febb4d51e..., Truncate length: 256}
```
Do you mean that it should count unique pair of Crypt and Auth?

Reply:
Yes. One way to achieve that would be to concatenate them with a separator before using that as the key in the set.

Reply:
(It needs a comment if that's the way you achieve this given it won't be obvious what/why we're doing from reading the code.)

Reply:
Gotcha, tnx! Do you think it might be possible to have `s.Auth != nil && s.Crypt == nil` (or vice versa)?

Reply:
I don't know. The best solution might be to error on such cases to avoid reporting incorrect results.

Reply:
Done.